applications/StatisticsApplication/custom_methods/spatial_methods.h

```diff
@@ -0,0 +1,883 @@
+        py::module non_historical_nodal_module =                                                         \
+            (py::module)non_historical_module.attr("Nodes");                                             \
+        py::module non_historical_nodal_norm_module =                                                    \
+            (py::module)non_historical_nodal_module.attr("NormMethods");                                 \
+        py::module non_historical_condition_module =                                                     \
+            (py::module)non_historical_module.attr("Conditions");                                        \
+        py::module non_historical_condition_norm_module =                                                \
+            (py::module)non_historical_condition_module.attr("NormMethods");                             \
+        py::module non_historical_element_module =                                                       \
+            (py::module)non_historical_module.attr("Elements");                                          \
+        py::module non_historical_element_norm_module =                                                  \
+            (py::module)non_historical_element_module.attr("NormMethods");                               \
+        ADD_KRATOS_STATISTICS_SPATIAL_CONTAINER_NORM_METHOD_PYTHON_INTERFACE(                            \
+            method, method_name, historical_norm_module, HistoricalSpatialMethods)                       \
+        ADD_KRATOS_STATISTICS_SPATIAL_CONTAINER_NORM_METHOD_PYTHON_INTERFACE(                            \
+            method, method_name, non_historical_nodal_norm_module, NodalNonHistoricalSpatialMethods)     \
+        ADD_KRATOS_STATISTICS_SPATIAL_CONTAINER_NORM_METHOD_PYTHON_INTERFACE(                            \
+            method, method_name, non_historical_condition_norm_module,                                   \
+            ConditionNonHistoricalSpatialMethods)                                                        \
+        ADD_KRATOS_STATISTICS_SPATIAL_CONTAINER_NORM_METHOD_PYTHON_INTERFACE(                            \
+            method, method_name, non_historical_element_norm_module, ElementNonHistoricalSpatialMethods) \
+    }
+#endif
+
+namespace Kratos
+{
+///@addtogroup RANSApplication
+///@{
+
+///@name Kratos Globals
+///@{
+
+namespace SpatialMethods
+{
+template <typename TContainerType, typename TContainerItemType, template <typename T> typename TDataRetrievalFunctor>
+class ContainerSpatialMethods
+{
+public:
+    // special overloaded method for flags
+    int static CalculateSum(const ModelPart& rModelPart, const Flags& rVariable)
+    {
+        const TContainerType& r_container =
+            MethodUtilities::GetLocalDataContainer<TContainerType>(rModelPart);
+        int sum = 0;
+
+#pragma omp parallel for reduction(+ : sum)
```

Comment:
I am pretty Win does not support reductions :/

Also I would indent the pragma with the same indentation as the surrounding code

Reply:
I thought only min max reductions are not supported by win? I used +, - reductions in RANSApplication and they worked with students. Should I be using my own +,- reductions as well?

Reply:
ok then its fine